flow-server/src/main/java/com/vaadin/flow/server/communication/rpc/AbstractRpcInvocationHandler.java

```diff
@@ -72,6 +77,110 @@ public Optional<Runnable> handle(UI ui, JsonObject invocationJson) {
         }
     }
 
+    /**
+     * Specifies whether inert status should be ignored for an RPC invocation or
+     * not. The default behaviour is to let the polling events be handled, while
+     * ignoring other requests.
+     *
+     * @param ui
+     *            the current UI instance
+     * @param invocationJson
+     *            the JsonObject containing invocation properties
+     * @return a boolean indicating that the inert status should be ignored for
+     *         the current invocation or not.
+     */
+    protected boolean ignoreInertForRpcInvocation(UI ui,
+            JsonObject invocationJson) {
+
+        if (!isPollEventInvocation(invocationJson)) {
+            return false;
+        }
+
+        if (!isPollingEnabledForUI(ui)) {
+            StringBuilder warning = new StringBuilder(
+                    "Ignoring Poll RPC for UI that does not have polling enabled.");
+            if (!VaadinService.getCurrent().getDeploymentConfiguration()
+                    .isProductionMode() && ui.getUIId() > 0) {
+                warning.append(" UI id: '{}'");
+            }
+            getLogger().warn(warning.toString(), ui.getUIId());
+            return false;
```

Comment:
```suggestion
            getLogger().warn("Ignoring Poll RPC for UI that does not have polling enabled.");
            getLogger().debug("Ignored payload\n{}", invocationJson);
            return false;
```
I doubt you can figure anything out with the UI.id so that is a bit extra. But for debugging the whole invocation could be added.

Reply:
Although the `debug` level of logging should not be enabled in Production mode normally, I wrapped the `getLogger().debug` within the condition of being in "Development" mode. Also, added tests to verify the level of logging in Production and Development modes.

Reply:
I would in this case say that the payload is much more interesting in production mode than dev mode as in production mode it means we are probably running a live server which someone is trying to send an unaccepted payload to and this shouldn't contain any information at all that would be non loggable for production mode.

Just to keep the feature clean I would drop anything you deem would need cheking execution mode for.

Reply:
You are right, logging the ignored payload would not create a security risk as even if the attacker can access the server logs they would see nothing but their own malicious payload and no sensitive data.

Removed everything related to the execution mode. Changed the tests as well.